src/otx/core/data/entity/keypoint_detection.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for OTX detection data entities."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from torchvision import tv_tensors
+
+from otx.core.data.entity.base import (
+    OTXBatchDataEntity,
+    OTXBatchPredEntity,
+    OTXDataEntity,
+    OTXPredEntity,
+)
+from otx.core.data.entity.utils import register_pytree_node
+from otx.core.types.task import OTXTaskType
+
+if TYPE_CHECKING:
+    from torch import BoolTensor, LongTensor
+
+
+@register_pytree_node
+@dataclass
+class KeypointDetDataEntity(OTXDataEntity):
+    """Data entity for keypoint detection task.
+
+    :param bboxes: Bbox annotations as top-left-bottom-right
+        (x1, y1, x2, y2) format with absolute coordinate values
+    :param labels: Bbox labels as integer indices
+    :param keypoints: keypoint annotations
+        ([[x1, y1], [x2, y2], ...]) format with absolute coordinate values
+    :param keypoints_visible: keypoint visibilities with binary values
+    :param keypoint_x_labels: x-axis keypoint coordinates according to simcc
+    :param keypoint_y_labels: y-axis keypoint coordinates according to simcc
+    :param keypoint_weights: weight values for each keypoint
+    """
+
+    @property
+    def task(self) -> OTXTaskType:
+        """OTX Task type definition."""
+        return OTXTaskType.KEYPOINT_DETECTION
+
+    bboxes: tv_tensors.BoundingBoxes
+    labels: LongTensor
+    keypoints: tv_tensors.TVTensor
+    keypoints_visible: BoolTensor
+    keypoint_x_labels: tv_tensors.TVTensor
+    keypoint_y_labels: tv_tensors.TVTensor
+    keypoint_weights: tv_tensors.TVTensor
```

Comment:
How about integrating `keypoint_x_labels`, `keypoint_y_labels`, and `keypoint_weights` (and `keypoints_visible` if possible?) to handle them at the same time using dataclass like below?
https://github.com/openvinotoolkit/training_extensions/blob/0f08b405b994ed5208c97d7388aa121c89e3e2b1/src/otx/core/data/entity/visual_prompting.py#L130-L136

Reply:
Let's discuss this in the near future.